### ISSUE TITLE
Low bandwidth and AES encryption

### ISSUE THREAD

**Issue by abarghou1993** [ntop/n2n#299]
I used the n2n-benchmark to test the n2n bandwidth in docker container. For no encryption I had a bandwidth of 3 Gbps, for AES encryption i had 150 Mbps. When i tested the bandwidth for data transfer between two edge nodes running in docker containers, I had 150 Mbps bandwidth even thou I disabled AES encryption.

**Comment by Logan007** [comment]
The `n2n-benchmark` tool performs encryption and optionally decryption in memory to give a rough idea on how the _ciphers_ perform. Thus, with `null_transform` it is just copying from one memory location to another. So, the throughput of 3 Gbps might resemble more the memory bandwidth than the actually possible network throughput.

`n2n-benchmark` completely omits the network and packet handling and supernode parts which would happen at both ends of the line. Speaking of two ends of the line, please note that `n2n-benchmark` only performs encryption by default. If run with `-d` command line option, it also performs decryption and therefore will output smaller numbers.

Long story short: Currently, `n2n-benchmark` shows cipher's performance but does not measure full-stack end-to-end network performance – which depends on a lot of additional factors of course.

**Comment by abarghou1993** [comment]
Thank you for your reply,
Do you know what is the typical bandwidth for end to end network, where the nodes run in docker containers? Also, do you know if the performance is lower if n2n runs on Azure Cloud Linux virtual machine.

**Comment by abarghou1993** [comment]
Thank you for your reply,
Do you know what is the typical bandwidth for end to end network, where the
nodes run in docker containers? Also, do you know if the performance is
lower if n2n runs on Azure Cloud Linux virtual machine.

**Comment by Logan007** [comment]
> Do _you_ know what is the typical bandwidth […]

I do not. But maybe others.

Although, I guess that it depends on a lot of parameters such as CPU power, memory bandwidth, allotted resources …

**Comment by abarghou1993** [comment]
The bandwidth on host with n2n is 900 Mbps. The bandwidth on host without
n2n is  4 Gbps. The bandwidth when running in docker container with n2n is
only 150 Mbps.

Thank you,
Abdel

**Comment by lucktu** [comment]
Maybe it has something to do with the driver of the virtual network card?

Plus, you can check out zerotier and tinc. What about their bandwidth?

**Comment by Logan007** [comment]
Any news?
